editor/plugins/parallax_background_editor_plugin.cpp

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "parallax_background_editor_plugin.h"
+
+#include "canvas_item_editor_plugin.h"
+#include "editor/editor_node.h"
+#include "editor/editor_undo_redo_manager.h"
+#include "editor/scene_tree_dock.h"
+#include "scene/2d/parallax_2d.h"
+#include "scene/2d/parallax_background.h"
+#include "scene/2d/parallax_layer.h"
+#include "scene/gui/box_container.h"
+#include "scene/gui/menu_button.h"
+
+void ParallaxBackgroundEditorPlugin::edit(Object *p_object) {
+	parallax_background = Object::cast_to<ParallaxBackground>(p_object);
+}
+
+bool ParallaxBackgroundEditorPlugin::handles(Object *p_object) const {
+	return Object::cast_to<ParallaxBackground>(p_object) != nullptr;
+}
+
+void ParallaxBackgroundEditorPlugin::make_visible(bool p_visible) {
+	if (p_visible) {
+		toolbar->show();
+	} else {
+		toolbar->hide();
+	}
+}
+
+void ParallaxBackgroundEditorPlugin::_menu_callback(int p_idx) {
+	if (p_idx == MENU_CONVERT_TO_PARALLAX_2D) {
+		convert_to_parallax2d();
+	}
+}
+
+void ParallaxBackgroundEditorPlugin::convert_to_parallax2d() {
+	ParallaxBackground *parallax_bg = parallax_background;
+	TypedArray<Node> children = parallax_bg->get_children();
+
+	EditorUndoRedoManager *ur = EditorUndoRedoManager::get_singleton();
+	ur->create_action(TTR("Convert to Parallax2D"));
```

Comment:
```suggestion
	ur->create_action(TTR("Convert to Parallax2D"), UndoRedo::MERGE_DISABLE, parallax_bg);
```
This fixes the problem with multiple undo actions.